### PR TITLE
[TEST] Add mock doublevalues script

### PR DESCRIPTION
**Pull request by grcevski**
Add mock DoubleValues script for the MockScriptEngine.

Relates to https://github.com/elastic/elasticsearch/pull/89895

### CODE REVIEW

File: test/framework/src/main/java/org/elasticsearch/script/MockScriptEngine.java [hunk 1]
```diff
@@ -339,6 +343,9 @@ public void execute() {
                 }
             };
             return context.factoryClazz.cast(objectFieldScript);
+        } else if (context.instanceClazz.equals(DoubleValuesScript.class)) {
+            DoubleValuesScript.Factory doubleValuesScript = () -> new MockDoubleValuesScript();
```
Comment (stu-elastic) [inline test/framework/src/main/java/org/elasticsearch/script/MockScriptEngine.java:347]: Can you add something that exercises this code?

Comment (grcevski) [reply test/framework/src/main/java/org/elasticsearch/script/MockScriptEngine.java:347]: Sure

Comment (grcevski) [reply test/framework/src/main/java/org/elasticsearch/script/MockScriptEngine.java:347]: I added a simple test, but also most of the tests in ScriptServiceTests will hit this code.

**Review by stu-elastic — APPROVED** [review]
lgtm

**Pull request merged**